Makefile

```diff
@@ -24,37 +31,54 @@ GOOGLEAPIS_RPC_CCS = $(shell find $(GOOGLEAPIS_GENS_PATH)/google/rpc \
 
 GOOGLEAPIS_CCS = $(GOOGLEAPIS_API_CCS) $(GOOGLEAPIS_RPC_CCS) $(GOOGLEAPIS_TYPE_CCS)
 
-GOOGLEAPIS_ASSISTANT_CCS = ./googleapis/gens/google/assistant/embedded/v1alpha2/embedded_assistant.pb.cc \
-	./googleapis/gens/google/assistant/embedded/v1alpha2/embedded_assistant.grpc.pb.cc
+GOOGLEAPIS_ASSISTANT_CCS = $(GOOGLEAPIS_GENS_PATH)/$(GOOGLEAPIS_ASSISTANT_PATH)/embedded_assistant.pb.cc \
+                           $(GOOGLEAPIS_GENS_PATH)/$(GOOGLEAPIS_ASSISTANT_PATH)/embedded_assistant.grpc.pb.cc
 
 HOST_SYSTEM = $(shell uname | cut -f 1 -d_)
 SYSTEM ?= $(HOST_SYSTEM)
-CXX = g++
-CPPFLAGS += -I/usr/local/include -pthread -I$(GOOGLEAPIS_GENS_PATH) \
-	    -I$(GRPC_SRC_PATH) -I./src/
-CXXFLAGS += -std=c++11
+
+HAS_PKG_CONFIG ?= true
+ifeq ($(HAS_PKG_CONFIG),true)
+GRPC_GRPCPP_CFLAGS=`pkg-config --cflags grpc++ grpc`
+GRPC_GRPCPP_LDLAGS=`pkg-config --libs grpc++ grpc`
+ALSA_CFLAGS=`pkg-config --cflags alsa`
+ALSA_LDFLAGS=`pkg-config --libs alsa`
+else
+GRPC_GRPCPP_CFLAGS ?=
```

Comment:
Would we ever not want grpc to be compiled? ALSA is Linux-only, but grpc is a core part of the sample.

Reply:
If you don't have pkg-config, you need a way to pass the CFLAGS/LDFLAGS needed to use gRPC, e.g. in my actual build script:

<pre>
    <...>
    local GRPC_GRPCPP_CFLAGS="-pthread -I${TARGET_INSTALL_PATH}/include"
    local GRPC_GRPCPP_LDLAGS="-L${TARGET_INSTALL_PATH}/lib -lgpr -laddress_sorting -lgrpc -lgrpc++"
    <...>
    GRPC_SRC_PATH=${GRPC_SRC_PATH} \
    GOOGLEAPIS_SRC_PATH=${GOOGLEAPIS_SRC_PATH} \
    GOOGLEAPIS_GENS_PATH=${GOOGLEAPIS_GENS_PATH} \
    HAS_PKG_CONFIG=false \
    SYSTEM=${SYSTEM} \
    GRPC_GRPCPP_CFLAGS=${GRPC_GRPCPP_CFLAGS} \
    GRPC_GRPCPP_LDLAGS=${GRPC_GRPCPP_LDLAGS} \
    CPPFLAGS=${CPPFLAGS} \
    LDFLAGS=${LDFLAGS} \
    CXX=${CXX} \
    AR=${AR} \
    PROTOC=${PROTOC} \
    PROTOC_PLUGINS_PATH=${PROTOC_PLUGINS_PATH} \
    make -C "${PROJECT_PATH}/assistant-sdk-cpp" clean run_assistant
</pre>

It's a way to help people with custom build systems without changing anything for the other users.